### ISSUE TITLE
Accessing gradient information (prev_delta_W) and storing logging  inside the snapshot data structure

### ISSUE THREAD

**Issue by alex-ten** [alex-ten/pdpyflow#5]
Jay, if you have time could you clarify what kind of data do we need to access / store here?
The following might be helpful:

1) We are using `tf.MomentumOptimizer` which updates parameters by the following rule:
  ```
  accumulation = momentum * accumulation + gradient
  variable -= learning_rate * accumulation
  ```
2) It inherits from the generic `Optimizer` class which includes a `minimize()` method
3) `minimize()` is a relatively high-level method which consists of `compute_gradients()` and `apply_gradients()`
4) I reckon we are interested in `compute_gradients()`, which at a lower level calls the `tf.gradients()` function, which is exactly what I am using to access gradient information upon testing

`tf.gradients()` is defined [here](https://www.tensorflow.org/api_docs/python/tf/gradients)
TL;DR: 

```
gradients(
    ys,
    xs,
    grad_ys=None,
    name='gradients',
    colocate_gradients_with_ops=False,
    gate_gradients=False,
    aggregation_method=None,
    stop_gradients=None
)
```
Constructs symbolic derivatives of sum of `ys` w.r.t. x in `xs`.

`ys` and `xs` are each a Tensor or a list of tensors. `grad_ys` is a list of `Tensor`, holding the gradients received by the `ys`. The list must be the same length as `ys`.

`gradients()` adds ops to the graph to output the derivatives of ys with respect to `xs`. It returns a list of Tensors of length `len(xs)` where each tensor is the `sum(dy/dx)` for `y` in `ys`.
